standard/io/docs.md

```diff
@@ -0,0 +1,1366 @@
+Executable file format error.
+
+- <a href="#errno.nolck" name="errno.nolck"></a> `nolck`
+No locks available.
+
+- <a href="#errno.nolink" name="errno.nolink"></a> `nolink`
+Reserved.
+
+- <a href="#errno.nomem" name="errno.nomem"></a> `nomem`
+Not enough space.
+
+- <a href="#errno.nomsg" name="errno.nomsg"></a> `nomsg`
+No message of the desired type.
+
+- <a href="#errno.noprotoopt" name="errno.noprotoopt"></a> `noprotoopt`
+Protocol not available.
+
+- <a href="#errno.nospc" name="errno.nospc"></a> `nospc`
+No space left on device.
+
+- <a href="#errno.nosys" name="errno.nosys"></a> `nosys`
+Function not supported.
+
+- <a href="#errno.notconn" name="errno.notconn"></a> `notconn`
+The socket is not connected.
+
+- <a href="#errno.notdir" name="errno.notdir"></a> `notdir`
+Not a directory or a symbolic link to a directory.
+
+- <a href="#errno.notempty" name="errno.notempty"></a> `notempty`
+Directory not empty.
+
+- <a href="#errno.notrecoverable" name="errno.notrecoverable"></a> `notrecoverable`
+State not recoverable.
+
+- <a href="#errno.notsock" name="errno.notsock"></a> `notsock`
+Not a socket.
+
+- <a href="#errno.notsup" name="errno.notsup"></a> `notsup`
+Not supported, or operation not supported on socket.
+
+- <a href="#errno.notty" name="errno.notty"></a> `notty`
+Inappropriate I/O control operation.
+
+- <a href="#errno.nxio" name="errno.nxio"></a> `nxio`
+No such device or address.
+
+- <a href="#errno.overflow" name="errno.overflow"></a> `overflow`
+Value too large to be stored in data type.
+
+- <a href="#errno.ownerdead" name="errno.ownerdead"></a> `ownerdead`
+Previous owner died.
+
+- <a href="#errno.perm" name="errno.perm"></a> `perm`
+Operation not permitted.
+
+- <a href="#errno.pipe" name="errno.pipe"></a> `pipe`
+Broken pipe.
+
+- <a href="#errno.proto" name="errno.proto"></a> `proto`
+Protocol error.
+
+- <a href="#errno.protonosupport" name="errno.protonosupport"></a> `protonosupport`
+Protocol not supported.
+
+- <a href="#errno.prototype" name="errno.prototype"></a> `prototype`
+Protocol wrong type for socket.
+
+- <a href="#errno.range" name="errno.range"></a> `range`
+Result too large.
+
+- <a href="#errno.rofs" name="errno.rofs"></a> `rofs`
+Read-only file system.
+
+- <a href="#errno.spipe" name="errno.spipe"></a> `spipe`
+Invalid seek.
+
+- <a href="#errno.srch" name="errno.srch"></a> `srch`
+No such process.
+
+- <a href="#errno.stale" name="errno.stale"></a> `stale`
+Reserved.
+
+- <a href="#errno.timedout" name="errno.timedout"></a> `timedout`
+Connection timed out.
+
+- <a href="#errno.txtbsy" name="errno.txtbsy"></a> `txtbsy`
+Text file busy.
+
+- <a href="#errno.xdev" name="errno.xdev"></a> `xdev`
+Cross-device link.
+
+- <a href="#errno.notcapable" name="errno.notcapable"></a> `notcapable`
+Extension: Capabilities insufficient.
+
+## <a href="#rights" name="rights"></a> `rights`: `Record`
+File descriptor rights, determining which actions may be performed.
```

Comment:
Do we need add a rights_extended for future usage?

Reply:
It's a good idea, however it turns out the rights system [isn't very useful](https://github.com/WebAssembly/WASI/issues/409#issuecomment-795494899), so it seems more likely that we'll just remove it altogether.